utils/das_data_sorter.py

```diff
@@ -1,76 +1,170 @@
 import pandas as pd 
-import argparse 
-from math import ceil
+from argparse import ArgumentParser
+from numpy import ceil
+from typing import TypeVar
 
-# averages data within the seconds, according to seconds array 
-def avg_Data(array):
-    avg_array = []
+D = TypeVar('DAS_data') # data type to represent columns of the data in a given file
```

Comment:
I really like the concept of this! Can we change the name `D` into a more descriptive type?

Reply:
I got rid of this hahaha. I used this to represent columns of data points, but turns out they have their own data type as discovered by Chris. 

Just copied this concept from my FIT1008 classes, but still unsure how to fully use this. Might ask my lecturer before utilising this.

Reply:
So according to my lecturer, using TypeVar is like using pronumerals in maths. For a math function like; f: R->R, f(x)=x^2+x+3, this function only allows real numbers to substitute for x. But in Python, TypeVar makes it so that x must be of the same data type throughout the script. 

If the Python function is `def function(arg1: T, arg2 :T)` then that means the argument passed by both arg1 and arg2 must both be strings. Or in another case, both must be integers. Depends on how the user wants to use the class. This ensures that the user is using the same data types for the class. 

For example, we cant do `function("str", 48)`, because T is confused whether it is a string or an integer. But we can do `function("str", "str")` or `function(48, 32)`. 

Upon saying that, she suggests that you only use TypeVar if the type to be used for all scripts is unknown. Otherwise, there's no point in using TypeVar if the arguments passed are intended for a specific data type, because you can just do `def function(arg1: int, arg2: int)`.